src/main/java/com/dabsquared/gitlabjenkins/trigger/handler/merge/MergeRequestHookTriggerHandlerImpl.java

```diff
@@ -196,10 +196,9 @@ protected CauseData retrieveCauseData(MergeRequestHook hook) {
                 .withTargetProjectId(hook.getObjectAttributes().getTargetProjectId())
                 .withBranch(hook.getObjectAttributes().getSourceBranch())
                 .withSourceBranch(hook.getObjectAttributes().getSourceBranch())
-                .withUserName(
-                        hook.getObjectAttributes().getLastCommit().getAuthor().getName())
-                .withUserEmail(
-                        hook.getObjectAttributes().getLastCommit().getAuthor().getEmail())
+                .withUserName(hook.getUser().getName())
+                .withUserUsername(hook.getUser().getUsername())
+                .withUserEmail(hook.getUser().getEmail())
```

Comment:
I am a bit worried these three changes have effects that are not intended.

Reply:
Yeah, I was also not sure about the change of these three attributes, but these values are the real ones that are sent by GitLab webhook, and it can be a little bit confusing that gitlabUserName and gitlabUserUsername can represent two different users. Author of the MR is not always the same person as author of the last commit, and these are two different attributes that comes from webhook.
Maybe it's good idea to set new variables called gitlabLastCommitAuthorName and gitlabLastCommitAuthorEmail. Just this two, as there are not the value for commiter username. But I know it still can impact someone's pipelines, so let me know if you think it's possible to make this type of changes.